### PR TITLE
[SPARK-22142][BUILD][STREAMING] Move Flume support behind a profile

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Add 'flume' profile to enable Flume-related integration modules

## How was this patch tested?

Existing tests; no functional change

### CODE REVIEW

File: python/pyspark/streaming/tests.py [hunk 1]
```diff
@@ -1478,7 +1478,7 @@ def search_kafka_assembly_jar():
             ("Failed to find Spark Streaming kafka assembly jar in %s. " % kafka_assembly_dir) +
             "You need to build Spark with "
             "'build/sbt assembly/package streaming-kafka-0-8-assembly/assembly' or "
-            "'build/mvn package' before running this test.")
+            "'build/mvn -Pkafka-0-8 package' before running this test.")
```
Comment (srowen) [inline python/pyspark/streaming/tests.py]: Not strictly related, but I noticed this when making a similar modification in the message below

**Comment by srowen** [comment]
Merged to master

**Comment by gatorsmile** [comment]
A few test cases failed. 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/82324/console

Could we revert this first?

**Comment by gatorsmile** [comment]
Just revert it back. Please submit a new PR. Thanks!

**Comment by srowen** [comment]
Hm, not sure why it passed Maven and the SBT PR builder. Yeah I'll have to look into it.

File: python/pyspark/streaming/tests.py (hunk 1 above)
Comment (apsist025350) [reply python/pyspark/streaming/tests.py]: LINE

**Comment by apsist025350** [comment]
[dropbox-airwatch-onepager.pdf](https://github.com/apache/spark/files/1381220/dropbox-airwatch-onepager.pdf)

**Pull request closed without merging**